### ISSUE TITLE
Consider allowing config parameters to be set after initialization

### ISSUE THREAD

**Issue by daattali** [daattali/dash2#4]
For example the parameters 'url_base_pathname', 'routes_pathname_prefix', 'requests_pathname_prefix' currently need to be set in the creation of the dash object (or in envvars). To me they don't sound like they should be associated with the function that *creates* the Dash object, because it couples the dash object right from the beginning to how it's going to be hosted. That feels like something that should either be added into the Dash object after creation, or provided to the app$run_server() function. Perhaps some sort of a "config" system, where config parameters can be added either in the constructor or in run_serve() or any time in between using app$modify_config().

Just some thoughts for later.

**Comment by daattali** [comment]
Full list of parameters that are currently in `Dash$new()` and would be candidates to set post-init:

- assets_folder
- assets_url_path
- assets_ignore
- eager_loading
- serve_locally
- url_base_pathname
- routes_pathname_prefix
- requests_pathname_prefix
- compress
- suppress_callback_exceptions
- show_undo_redo

**Comment by alexcjohnson** [comment]
Good idea - FWIW on the Python side we do allow changing config afterward, like `app.config.title="My App"`, but with some of the attributes locked after instantiating the app. If anyone complained we could likely find ways to relax these locks, but I haven't seen late setting get much use in the wild let alone requests to relax the locked attributes. That said we haven't promoted this in the docs, so it's quite possible folks would use it if they knew it existed!

https://github.com/plotly/dash/blob/efc42b46fef033bd4ec1f082cfe5cbab97a62646/dash/dash.py#L305-L352

**Comment by daattali** [comment]
Interesting, that would be useful in my opinion to publicize that there's a config that can be modified.
